google-cloud-logging/lib/google/cloud/logging/middleware.rb

```diff
@@ -74,6 +74,56 @@ def extract_trace_id env
         end
 
         ##
+        # Construct a monitored resource based on the given type and label if
+        # both are provided. Otherwise, construct a default monitored resource
+        # based on the current environment.
+        #
+        # If not given both type and label:
+        #   If running from GAE, return resource:
+        #   {
+        #     type: "gae_app", {
+        #       module_id: [GAE module name],
+        #       version_id: [GAE module version]
+        #     }
+        #   }
+        #   If running from GKE, return resource:
+        #   {
+        #     type: "container", {
+        #       cluster_name: [GKE cluster name],
+        #       namespace_id: [GKE namespace_id]
+        #     }
+        #   }
+        #   If running from GCE, return resource:
+        #   {
+        #     type: "gce_instance", {
+        #       instance_id: [GCE VM instance id],
+        #       zone: [GCE vm group zone]
+        #     }
+        #   }
+        #   Otherwise default to { type: "global" }, which means not associated
+        #   with GCP.
```

Comment:
What do you think about using YARD's `@example` for each of these? And showing Ruby code instead of the underlying API JSON? (The actual return type is a Ruby object, `Resource`, not JSON, correct?)